filter.go

```diff
@@ -1,11 +1,73 @@
 package notionapi
 
+import (
+	"encoding/json"
+)
+
 type FilterOperator string
 
+type Filter interface {
+	filter()
+}
+
 type CompoundFilter map[FilterOperator][]PropertyFilter
 
+type AndCompoundFilter []Filter
+type OrCompoundFilter []Filter
+
+func (f AndCompoundFilter) filter() {}
+func (f OrCompoundFilter) filter()  {}
+
+func (f AndCompoundFilter) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		And []Filter `json:"and"`
+	}{
+		And: f,
+	})
+}
+
+func (f OrCompoundFilter) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		Or []Filter `json:"or"`
+	}{
+		Or: f,
+	})
+}
+
 type Condition string
 
+type TimestampCreatedTimeFilter struct {
```

Comment:
From my point of view the NotionAPI library should be as close to the API Reference as possible. Terms from the docs should be mapped to the library types directly(if it possible ofc).

That's why I suggest to make general `TimestampFilter` instead of  `TimestampCreatedTimeFilter` and `TimestampLastEditedTimeFilter`.  It'll make client's code a bit more verbose, but the library will be a bit easier to use.

WDYT?

Reply:
Ok, but I'd like to mention that officially notion-sdk-js [uses](https://github.com/makenotion/notion-sdk-js/blob/main/src/api-endpoints.ts#L8438) two different types. I used the library as a reference for implementation.